File: app/src/main/java/org/techtown/find_gas_station/Comparator/OilSpendTimeComparator.java
package org.techtown.find_gas_station.Comparator;

import org.techtown.find_gas_station.OilList;

import java.util.Comparator;

public class OilSpendTimeComparator implements Comparator<OilList> {
    @Override
    public int compare(OilList t1, OilList t2) {

        if(Integer.parseInt(t1.getSpendTime()) == Integer.parseInt(t2.getSpendTime())){
            return Integer.parseInt(t1.getPrice()) - Integer.parseInt(t2.getPrice());
        }
        else{
            return Integer.parseInt(t1.getSpendTime())-Integer.parseInt(t2.getSpendTime());
        }
    }


}

File: app/src/main/java/org/techtown/find_gas_station/Util/Comparator/OilSpendTimeComparator.kt
package org.techtown.find_gas_station.Util.Comparator

import org.techtown.find_gas_station.Data.TotalOilInfo


class OilSpendTimeComparator : Comparator<TotalOilInfo> {
    override fun compare(t1 : TotalOilInfo, t2 : TotalOilInfo) =
        if (t1.spendTime.toInt() == t2.spendTime.toInt()) {
            t1.price.toInt() - t2.price.toInt()
        } else {
            t1.spendTime.toInt() - t2.spendTime.toInt()
        }

}